mirgecom/__init__.py

```diff
@@ -19,3 +19,43 @@
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 """
+
+import os
+import getpass
+import sys
+import tempfile
+from warnings import warn
+
+from mpi4py import MPI
+
+# This code avoids slow startups due to file system locking when running with
+# large numbers of ranks. See https://github.com/illinois-ceesd/planning/issues/27
+# for details.
+
+size = MPI.COMM_WORLD.Get_size()
+rank = MPI.COMM_WORLD.Get_rank()
+
+if size >= 256 and rank == 0:
+    if "XDG_CACHE_HOME" not in os.environ:
+        skip = False
+
+        for mod in ["pyopencl", "pytools", "loopy"]:
+            if mod in sys.modules:
+                warn(f"{mod} already loaded, not adjusting XDG_CACHE_HOME. Please "
+                  "see https://github.com/illinois-ceesd/planning/issues/27 for "
```

Comment:
This is not a publicly accessible URL. Could you add a section to the docs explaining this and link to that?

(same above)

Reply:
Added to docs.